lib/ble.toit

```diff
@@ -257,23 +270,157 @@ class Client:
     state := resource_state_.wait_for_state STARTED_EVENT_
     resource_state_.clear_state STARTED_EVENT_
 
+class ServerConfig:
+  resource_group_ := null
+  services/List := []
+
+  constructor:
+    resource_group_ = ble_server_config_init_
+
+  add_service uuid -> Service:
+    service := Service resource_group_ uuid
+    services.add service
+    return Service resource_group_ uuid
+
+
+/** 
+Defines a BLE service with characteristics 
+*/
+class Service:
+  /**
+  The UUID of the service.
+  
+  For 16 and 32 bit UUIDs, form the BLE variant with the top-level uuid function.
+  */
+  uuid/uuid_pkg.Uuid
+  
+  resource_group_/any
+  resource_/any
+
+  characteristics_/Map := {:}
+
+  constructor .resource_group_ .uuid:
+    resource_ = ble_add_server_service_ resource_group_ uuid.to_byte_array
+
+  add_read_only_characteristic uuid --value=#[]-> ReadOnlyCharacteristic:
+    return ReadOnlyCharacteristic this uuid value
+
+  add_write_only_characteristic uuid -> WriteOnlyCharacteristic:
+    return WriteOnlyCharacteristic this uuid
+
+  add_read_write_characteristic uuid --value=#[]-> ReadWriteCharacteristic:
+    return ReadWriteCharacteristic this uuid value
+
+  add_notification_characteristic uuid -> NotificationCharacteristic:
+    return NotificationCharacteristic this uuid
+
+  get_characteristic uuid -> Characteristic?:
+    return characteristics_[uuid]
```

Comment:
Is the intention that this will return null if the characteristic isn't found? You should use `characteristics_.get uuid` instead then, because the `[]` will throw if the `uuid` isn't in the map.